practices/libraries.md

```diff
@@ -0,0 +1,99 @@
+---
+slug: libraries
+stages:
+  - development
+short_description: Updating dependencies from your libraries can be a huge pain. But it is a necessary pain! Your whole project can be jeopardized without regular updates. Automatize and keep track of the dependencies’ changes.
+tags:
+  - libraries
+  - lockfile
+  - dependencies
+  - improvement
+  - functionality
+  - automated deployment
+  - continuous integration
+  - continuous delivery
+  - continuous deployment
+  - unit tests
+keywords:
+  - software library
+  - updating software
+  - updating dependencies
+  - updating libraries
+covered_by_dx_scanner: true
+---
+
+# Libraries
+
+**TL;DR**
+
+Updating dependencies from your libraries can be a huge pain. But it is a necessary pain! Your whole project can be jeopardized without regular updates. Automatize and keep track of the dependencies’ changes.
+
+## What Are the Libraries
+
+Libraries collect resources for software development. They consist of pre-written code, procedures, classes, scripts, data configurations, and more. Developers add libraries to the program they are developing to get some functionalities, dependencies, or automation without having to program it specifically for their program.
+
+Here are a few advantages in using libraries:
+
+- The library code can be used by multiple different programs. In comparison, the code that is part of the program can be used only within the program.
+- While using the library, the program gains the library’s dependencies and does not have to implement the dependencies itself.
+- The library’s properties can be connected to the program in any lifecycle phase.
+
+## Why You Might Want the Libraries to Be Updated
+
+Sometimes the updates are arriving almost every day and it can get very annoying. Updating is a pain. It takes a long time and after the update, nothing works as before. But remember - libraries updates might be very important. The dependencies from the specific library you need for your program to be working may change.
+
+You should keep track of those updates, they might contain important changes, for example:
+
+- Security vulnerabilities fixes
+- Bug fixes
+- Performance improvements
+- New functionalities
+- Better user interface
+
+You might need some of these changes urgently. Checking them regularly and updating them as soon as possible saves you time and money. You have to update the dependencies anyway, so why to postpone and, therefore, accumulate a maintenance debt?
```

Comment:
We should write that the most important is to update patch and minor versions (see semver.org) that contains the most important bug fixes here.